debs/gnome-control-center/puavo/patches/0011-puavo-appearance-make-color-scheme-switch-change-als.patch

```diff
@@ -0,0 +1,165 @@
+restarted to apply the color scheme change.
+
+I wonder if there's a DBus signal which we could emit?
+---
+ panels/background/cc-background-panel.c | 105 +++++++++++++++++++++++-
+ 1 file changed, 104 insertions(+), 1 deletion(-)
+
+diff --git a/panels/background/cc-background-panel.c b/panels/background/cc-background-panel.c
+index aa3c043ad..d2b2eaa3c 100644
+--- a/panels/background/cc-background-panel.c
++++ b/panels/background/cc-background-panel.c
+@@ -200,9 +200,102 @@ get_next_theme (const gchar         *theme,
+       g_warning("Dark theme does not exist for theme %s\n", theme);
+     }
+ 
++  return g_strdup (default_theme);
++}
+ 
++static GKeyFile*
++load_qt_config_key_file(const gchar *const config_file_path)
++{
++  g_autoptr(GError) error = NULL;
++  GKeyFile *key_file = g_key_file_new();
+ 
+-  return g_strdup (default_theme);
++  if (!g_key_file_load_from_file (key_file, config_file_path, G_KEY_FILE_NONE, &error))
++    {
++      if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
++        g_warning ("Failed to load Qt config file '%s': %s", config_file_path, error->message);
++      g_key_file_free(key_file);
++      return NULL;
++    }
++
++  return key_file;
++}
++
++static gchar*
++read_qt_config_theme(const gchar *const config_file_path)
++{
++  g_autoptr(GKeyFile) key_file = load_qt_config_key_file (config_file_path);
++
++  if (key_file == NULL)
++    return NULL;
++
++  return g_key_file_get_string (key_file, "General", "theme", NULL);
++}
++
++static gchar*
++get_qt_config_file_path(const gchar *const base_dir)
++{
++  return g_strconcat(base_dir, "/Kvantum/kvantum.kvconfig", NULL);
++}
++
++static gchar*
++get_qt_config_file_path_user()
++{
++  return get_qt_config_file_path (g_get_user_config_dir ());
++}
++
++static gchar*
++get_qt_theme_system()
++{
++  const gchar *const *dirs = g_get_system_config_dirs ();
++
++  for (int i = 0; dirs[i] != NULL; ++i)
++    {
++      gchar *config_file_path = get_qt_config_file_path (dirs[i]);
++      gchar *theme = read_qt_config_theme (config_file_path);
++
++      g_free (config_file_path);
++
++      if (theme)
++        return theme;
++    }
++
++  return NULL;
++}
++
++static gchar*
++get_qt_theme_user()
++{
++  g_autofree gchar *config_file_path = get_qt_config_file_path_user ();
++
++  return read_qt_config_theme (config_file_path);
++}
++
++static gchar*
++get_qt_theme ()
++{
++  gchar *theme;
++
++  theme = get_qt_theme_user ();
++  if (theme == NULL)
++    theme = get_qt_theme_system ();
++
++  return theme;
++}
++
++static void
++set_qt_theme (const gchar *theme)
++{
++  g_autoptr(GError) error = NULL;
++  g_autofree gchar *config_file_path = get_qt_config_file_path_user ();
++  g_autoptr(GKeyFile) key_file = load_qt_config_key_file (config_file_path);
++
++  if (key_file == NULL)
++    key_file = g_key_file_new ();
++
++  g_key_file_set_string (key_file, "General", "theme", theme);
++
++  if (!g_key_file_save_to_file (key_file, config_file_path, &error))
```

Comment:
Writing config file only works if ~/.config/Kvantum/ exists. Otherwise fails with error "Failed to save Qt config file '/home/user/.config/Kvantum/kvantum.kvconfig':' and a notice of failed temporary file creation.

Reply:
Yes, good, seems to work now. LGTM ( + added an optional note on live updates of the theme)